card/db/migrate_core_cards/20150429090551_search_card_context.rb

```diff
@@ -0,0 +1,25 @@
+# -*- encoding : utf-8 -*-
+
+class SearchCardContext < Card::CoreMigration
+  def up
+    sep = %r{\W}
+    replace = [
+      ['[lr]+','l\\1'],
+      ['[LR]+','L\\1'],
+      ["(?=[lrLR]+#{sep})(?=[LR]*[lr]+)(?=[lr]*[LR]+).*",'l\\1'],   # mix of lowercase and uppercase l's and r's
+      ['left',   'LL'],
+      ['right',  'LR'],
+      ['self',   'left'],
+      ['',       'left'],
+    ]
+    Card.search(:type=>'search').each do |card|
+      if card.cardname.junction?
+        content = card.content
+        replace.each do |key, val|
+          content.gsub!(/(?<=#{sep})_(#{key})(?=#{sep})/, "_#{val}")
```

Comment:
`(?<=...)` and `(?=...)` are the lookbehind and lookahead operators. They don't change the position of the regex engine and the pattern is not a part of the match.
In line 19 they make sure that there is no word character before or after the contextual name.
In line 9 the first lookahead finds contextual names containing [lrLR], the second assures that there is at least one uppercase letter and the third that there is at least one lowercase letter. 
This is necessary because I have to make sure that I don't process the completely lowercase and uppercase contextual names that I handled already in line 7 and 8.
I'm not sure if we support mixed cases though. I couldn't find the place where we process contextual names.